src/data/equiv/fin.lean

```diff
@@ -244,6 +244,42 @@ begin
     refl, }
 end
 
+@[simp] lemma fin_rotate_zero : fin_rotate 0 = equiv.refl _ := rfl
+
+@[simp] lemma fin_rotate_one : fin_rotate 1 = equiv.refl _ :=
+subsingleton.elim _ _
+
+lemma fin_rotate_succ_apply {n : ℕ} (i : fin n.succ) :
+  fin_rotate n.succ i = i + 1 :=
+begin
+  cases n,
+  { simp },
+  rcases i.le_last.eq_or_lt with rfl|h,
+  { simp },
+  { cases i,
+    simp only [fin.lt_iff_coe_lt_coe, fin.coe_last, fin.coe_mk] at h,
+    simp [fin_rotate_of_lt h, fin.eq_iff_veq, fin.add_def, nat.mod_eq_of_lt (nat.succ_lt_succ h)] },
+end
+
+@[simp] lemma fin_rotate_apply_zero {n : ℕ} : fin_rotate n.succ 0 = 1 :=
+by rw [fin_rotate_succ_apply, zero_add]
+
+lemma coe_fin_rotate_of_ne_last {n : ℕ} {i : fin n.succ} (h : i ≠ fin.last n) :
+  (fin_rotate n.succ i : ℕ) = i + 1 :=
+begin
+  rw fin_rotate_succ_apply,
+  have : (i : ℕ) < n := lt_of_le_of_ne (nat.succ_le_succ_iff.mp i.2) (fin.coe_injective.ne h),
+  exact fin.coe_add_one this
+end
+
+lemma coe_fin_rotate {n : ℕ} (i : fin n.succ) :
+  (fin_rotate n.succ i : ℕ) = if i = fin.last n then 0 else i + 1 :=
+begin
+  split_ifs with h,
+  { simp [fin_rotate_succ_apply, h] },
+  exact coe_fin_rotate_of_ne_last h
+end
```

Comment:
Does it make more sense to state this as
```suggestion
lemma coe_add_one {n : ℕ} (i : fin n.succ) :
  (i + 1 : ℕ) = if i = fin.last n then 0 else i + 1 :=
```
and then put it in `data/fin/basic`?

Reply:
Now that you've done that, can you prove this lemma using the new `coe_add_one`?